BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -3,19 +3,153 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
 import Foundation
+import UIKit
+import Common
+import Distributed
 
+// MARK: Protocol
 protocol TabDataStore {
-    func fetchTabData() async -> WindowData
-    func saveTabData(window: WindowData) async
-    func clearAllTabData() async
+    func fetchWindowData() async -> WindowData
+    func saveWindowData(window: WindowData) async
+    func clearAllWindowsData() async
 }
 
 actor DefaultTabDataStore: TabDataStore {
-    func fetchTabData() async -> WindowData {
+    // MARK: Variables
+    let browserKitInfo = BrowserKitInformation.shared
+    static let storePath = "codableWindowsState.archive"
+    static let profilePath = "profile.profile"
+    private var logger: Logger = DefaultLogger.shared
+
+    // MARK: URL Utils
+    private var windowDataDirectoryURL: URL? {
+        FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: browserKitInfo.sharedContainerIdentifier)?
+            .appendingPathComponent(DefaultTabDataStore.profilePath)
+    }
+
+    private func windowURLPath(for windowID: UUID) -> URL? {
+        if let profileURL = windowDataDirectoryURL {
+            let filePath = DefaultTabDataStore.storePath + "_\(windowID.uuidString)"
+            let windowProfileURL = profileURL.appendingPathComponent(filePath)
+            return windowProfileURL
+        }
+        return nil
+    }
+
+    // MARK: Fetching Window Data
+    func fetchWindowData() async -> WindowData {
         return WindowData(id: UUID(), isPrimary: true, activeTabId: UUID(), tabData: [])
     }
 
-    func saveTabData(window: WindowData) async {}
+    func fetchWindowData(withID id: UUID) async -> WindowData? {
+        guard let profileURL = self.windowURLPath(for: id) else {
+            return nil
+        }
+        do {
+
+            let data = try Data(contentsOf: profileURL)
+            let windowData = try JSONDecoder().decode(WindowData.self, from: data)
+            return windowData
+        } catch {
+            logger.log("Error decoding window data: \(error)",
+                       level: .debug,
+                       category: .tabs)
+        }
```

Comment:
I also just noticed this block of code is pretty much duplicated between here and the fetch all windows function. You could probably put this in a helper function and call it from within the loop in the fetch all function.